tests/main/views/test_create_user.py

```diff
@@ -121,26 +121,74 @@ def test_should_render_correct_error_page_if_token_expired(
                 t="send a new one",
             )
 
-    def test_should_render_create_user_page_if_user_does_not_exist(self):
+    def test_should_render_create_user_page_if_supplier_does_not_exist(self):
         self.data_api_client.get_user.return_value = None
-        page_titles = ["Create a new Digital Marketplace account", "Add your name and create a password"]
-        button_values = ['Create account'] * 2  # the same for now
 
-        for role, page_title, button_value in zip(self.user_roles, page_titles, button_values):
-            token = self._generate_token(role=role)
-            res = self.client.get(
-                '/user/create/{}'.format(token)
-            )
-            assert res.status_code == 200
+        token = self._generate_token(role="supplier")
+        res = self.client.get(f"/user/create/{token}")
+        assert res.status_code == 200
 
-            for message in [
-                page_title,
-                button_value,
-                "test@email.com",
-                '<form autocomplete="off" action="/user/create/%s" method="POST" id="createUserForm">'
-                    % urllib.parse.quote(token)
-            ]:
-                assert message in res.get_data(as_text=True)
+        doc = html.fromstring(res.get_data(as_text=True))
+
+        assert doc.cssselect(
+            "span:contains('test@email.com')"
+            "+ h1:contains('Add your name and create a password')"
```

Comment:
FWIW this form of content matching doesn't allow for anything like e.g. whitespace differences.

Reply:
Yeah, in this case though I want the check to be whitespace insensitive, and I think this is more readable